### PR TITLE
negate the result and prefix the metric name for error/loss metrics

### PR DESCRIPTION
**Pull request by sebhrusen**
https://github.com/openml/automlbenchmark/issues/268

Only changing the `result` and `metric` columns when the metric represents an error:
- `metric` name is prefixed with `neg_`.
- `result` is negated.

Example:
```text
Summing up scores for current run:
                  id         task          framework constraint fold    result       metric   mode version params               app_version                  utc  duration  training_duration  predict_duration models_count       seed       acc  auc    balacc   logloss      mae        r2     rmse
0  openml.org/t/3913          kc2  constantpredictor       test    0   0.50000          auc  local  0.23.2         dev [issue/268, b990e93]  2021-04-01T15:52:54       0.2                0.0               0.0            1  933769621  0.792453  0.5  0.500000  0.510714      NaN       NaN      NaN
1  openml.org/t/3913          kc2  constantpredictor       test    1   0.50000          auc  local  0.23.2         dev [issue/268, b990e93]  2021-04-01T15:52:54       0.1                0.0               0.0            1  933769622  0.792453  0.5  0.500000  0.510714      NaN       NaN      NaN
2    openml.org/t/59         iris  constantpredictor       test    0  -1.09861  neg_logloss  local  0.23.2         dev [issue/268, b990e93]  2021-04-01T15:52:54       0.0                0.0               0.0            1  933769621  0.333333  NaN  0.333333  1.098610      NaN       NaN      NaN
3    openml.org/t/59         iris  constantpredictor       test    1  -1.09861  neg_logloss  local  0.23.2         dev [issue/268, b990e93]  2021-04-01T15:52:54       0.0                0.0               0.0            1  933769622  0.333333  NaN  0.333333  1.098610      NaN       NaN      NaN
4  openml.org/t/2295  cholesterol  constantpredictor       test    0 -45.68970     neg_rmse  local  0.23.2         dev [issue/268, b990e93]  2021-04-01T15:52:54       0.0                0.0               0.0            1  933769621       NaN  NaN       NaN       NaN  35.6774 -0.077562  45.6897
5  openml.org/t/2295  cholesterol  constantpredictor       test    1 -55.00410     neg_rmse  local  0.23.2         dev [issue/268, b990e93]  2021-04-01T15:52:54       0.0                0.0               0.0            1  933769622       NaN  NaN       NaN       NaN  45.3871 -0.049133  55.0041
```

### CODE REVIEW

**Comment by sebhrusen** [comment]
@Innixma does this look reasonable?

**Review by PGijsbers — APPROVED** [review]
Prefixing `neg_` is probably more sensible (since scikit-learn does it) so I agree with that choice even though it takes more space. Other than that I tested it (with constant predictor) and it looks good.

File: amlb/results.py [hunk 1]
```diff
@@ -464,42 +472,80 @@ def __init__(self, predictions_df, info=None):
...
 
     def acc(self):
+        """Accuracy"""
         return float(accuracy_score(self.truth, self.predictions))
 
-    def balacc(self):
-        return float(balanced_accuracy_score(self.truth, self.predictions))
-
     def auc(self):
+        """Array Under (ROC) Curve, computed on probabilities, not on predictions"""
```
Comment (Innixma) [inline amlb/results.py:479]: nit: area instead of array

File: amlb/results.py [hunk 2]
```diff
@@ -510,24 +556,34 @@ def __init__(self, predictions_df, info=None):
...
     def rmsle(self):
+        """Root Mean Square Logarithmic Error"""
         return math.sqrt(self.msle())
 
     def r2(self):
+        """R^2"""
         return float(r2_score(self.truth, self.predictions))
 
 
+def higher_is_better(metric):
```
Comment (Innixma) [inline amlb/results.py:583]: This seems a bit hacky. Better to have either a dictionary mapping or metrics as classes ([example in AutoGluon](https://github.com/awslabs/autogluon/blob/master/core/src/autogluon/core/metrics/__init__.py#L17)).

**Review by Innixma — COMMENTED** [review]
Apologies for late response, I was on PTO.

Looks good to me, had a comment for long-term improving code quality and extensibility.

File: amlb/results.py (hunk 1 above)
Comment (sebhrusen) [reply amlb/results.py:479]: oups! will fix

File: amlb/results.py (hunk 2 above)
Comment (sebhrusen) [reply amlb/results.py:583]: I can't disagree with you: it IS a bit hacky.
Ideally, there should be a class for each metric. It's probably something I'll do at some point to support custom metrics or other customizations in a more satisfying way that what was done in https://github.com/openml/automlbenchmark/pull/141.
If there's a demand for it, I'll do it.

**Pull request merged**